### PR TITLE
Update stub/forward-host notation to accept port and tls-auth-name

### LINKED ISSUE #546: "forward-host" do NOT support TLS upstream
According to my viewing of code and experiments, "forward-host" config item dont support TLS.
SUPPORT:         forward-host: 1.1.1.1@853#one.one.one.one    
NOT SUPPORT:  forward-host:one.one.one.one@853#one.one.one.one

If I am right, can you please tell me are there any reasons why "forward-host" dont support TLS.
Or it is just to be done to support more features that are less important like this;  and if so, do we have any plan to support this?

### PR DESCRIPTION
**Pull request by gthess**
Fixes #546.

### CODE REVIEW

File: doc/unbound.conf.5.in [hunk 1]
```diff
@@ -1842,13 +1842,19 @@ Name of the stub zone. This is the full domain name of the zone.
 .TP
 .B stub\-host: \fI<domain name>
 Name of stub zone nameserver. Is itself resolved before it is used.
+To use a nondefault port for DNS communication append '@' with the port number.
+If tls is enabled, then you can append a '#' and a name, then it'll check the
+tls authentication certificates with that name.  If you combine the '@' and
+'#', the '@' comes first.  If only '#' is used the default port is the
```
Comment (wcawijngaards) [inline doc/unbound.conf.5.in]: The `'` at the start of the line indicates a troff request macro that formats the text, like the starting '.' on a line. So change it to not have that character at the start of the line.
```suggestion
tls authentication certificates with that name.  If you combine the '@'
and '#', the '@' comes first.  If only '#' is used the default port is the
```

File: doc/unbound.conf.5.in [hunk 2]
```diff
@@ -1842,13 +1842,19 @@ Name of the stub zone. This is the full domain name of the zone.
...
 .TP
 .B stub\-addr: \fI<IP address>
 IP address of stub zone nameserver. Can be IP 4 or IP 6.
 To use a nondefault port for DNS communication append '@' with the port number.
-If tls is enabled, then you can append a '#' and a name, then it'll check
-the tls authentication certificates with that name.  If you combine
-the '@' and '#', the '@' comes first.
+If tls is enabled, then you can append a '#' and a name, then it'll check the
+tls authentication certificates with that name.  If you combine the '@' and
+'#', the '@' comes first.  If only '#' is used the default port is the
```
Comment (wcawijngaards) [inline doc/unbound.conf.5.in]: Initial character is a troff macro
```suggestion
tls authentication certificates with that name.  If you combine the '@'
and '#', the '@' comes first.  If only '#' is used the default port is the
```

File: doc/unbound.conf.5.in [hunk 3]
```diff
@@ -1899,13 +1905,19 @@ Name of the forward zone. This is the full domain name of the zone.
 .TP
 .B forward\-host: \fI<domain name>
 Name of server to forward to. Is itself resolved before it is used.
+To use a nondefault port for DNS communication append '@' with the port number.
+If tls is enabled, then you can append a '#' and a name, then it'll check the
+tls authentication certificates with that name.  If you combine the '@' and
+'#', the '@' comes first.  If only '#' is used the default port is the
```
Comment (wcawijngaards) [inline doc/unbound.conf.5.in]: Initial character
```suggestion
tls authentication certificates with that name.  If you combine the '@'
and '#', the '@' comes first.  If only '#' is used the default port is the
```

File: doc/unbound.conf.5.in [hunk 4]
```diff
@@ -1899,13 +1905,19 @@ Name of the forward zone. This is the full domain name of the zone.
...
 .TP
 .B forward\-addr: \fI<IP address>
 IP address of server to forward to. Can be IP 4 or IP 6.
 To use a nondefault port for DNS communication append '@' with the port number.
-If tls is enabled, then you can append a '#' and a name, then it'll check
-the tls authentication certificates with that name.  If you combine
-the '@' and '#', the '@' comes first.
+If tls is enabled, then you can append a '#' and a name, then it'll check the
+tls authentication certificates with that name.  If you combine the '@' and
+'#', the '@' comes first.  If only '#' is used the default port is the
```
Comment (wcawijngaards) [inline doc/unbound.conf.5.in]: Initial character
```suggestion
tls authentication certificates with that name.  If you combine the '@'
and '#', the '@' comes first.  If only '#' is used the default port is the
```

**Review by wcawijngaards — APPROVED** [review]
The code looks good to me.

**Comment by gthess** [comment]
Thanks, good catch on the troff macro! I was wondering (but not too much apparently) why my text editor seemed to have different highlight at these points.

**Pull request merged**